File: rerun_py/rerun_sdk/rerun/archetypes/view_coordinates2d_ext.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

from ..components import ViewCoordinates2D as Component

if TYPE_CHECKING:
    from ..components import ViewCoordinates2D


class ViewCoordinates2DExt:
    """Extension for [ViewCoordinates2D][rerun.archetypes.ViewCoordinates2D]."""

    RD: ViewCoordinates2D = None  # type: ignore[assignment]
    """X=Right, Y=Down (default, image/screen convention)."""

    RU: ViewCoordinates2D = None  # type: ignore[assignment]
    """X=Right, Y=Up (math/plot convention)."""

    LD: ViewCoordinates2D = None  # type: ignore[assignment]
    """X=Left, Y=Down (horizontally mirrored image)."""

    LU: ViewCoordinates2D = None  # type: ignore[assignment]
    """X=Left, Y=Up (both axes flipped)."""

    @staticmethod
    def deferred_patch_class(cls: Any) -> None:
        cls.RD = Component.RD
        cls.RU = Component.RU
        cls.LD = Component.LD
        cls.LU = Component.LU

File: crates/store/re_sdk_types/src/archetypes/view_coordinates2d_ext.rs
#![expect(non_snake_case)]

use super::ViewCoordinates2D;
use crate::components;

impl ViewCoordinates2D {
    /// X=Right, Y=Down (default, image/screen convention).
    pub fn RD() -> Self {
        Self::new(components::ViewCoordinates2D::RD)
    }

    /// X=Right, Y=Up (math/plot convention).
    pub fn RU() -> Self {
        Self::new(components::ViewCoordinates2D::RU)
    }

    /// X=Left, Y=Down (horizontally mirrored image).
    pub fn LD() -> Self {
        Self::new(components::ViewCoordinates2D::LD)
    }

    /// X=Left, Y=Up (both axes flipped).
    pub fn LU() -> Self {
        Self::new(components::ViewCoordinates2D::LU)
    }
}
